backend/src/openarchiefbeheer/external_registers/registry.py

```diff
@@ -0,0 +1,43 @@
+from collections.abc import ItemsView
+from typing import Callable, Iterator
+
+from openarchiefbeheer.external_registers.plugin import AbstractBasePlugin
+
+
+class Registry[PluginT: AbstractBasePlugin]:
```

Comment:
Discussed with Chris: 
The `[PluginT: AbstractBasePlugin]` can be moved to the __call__ function.

And instead of being generic it can be only AbstractBasePlugin in the other places